rtos/ThisThread.h

```diff
@@ -174,7 +174,7 @@ void sleep_until(uint64_t millisec);
 void yield();
 
 /** Get the thread id of the current running thread.
-  @return  thread ID for reference by other functions or NULL in case of error or in ISR context.
+  @return  thread ID for reference by other functions or NULL in case of error.
```

Comment:
You may be right here - I think it actually returns current foreground thread if called from ISR.

Wasn't inconsistent though - documenting it as returning NULL from ISR is perfectly logical.

Saying it returns NULL from ISR but then saying you can't call it from ISR would be inconsistent.

Reply:
Although the new documented behaviour is maybe what you get from RTX `osThreadGetId`, I don't think it's actually what we would want from `ThisThread::get_id`, given the name. 

Fixing implementation to match documented behaviour would be better I think. Or specifying it can't be called from ISR.